src/sampleutil.cpp

```diff
@@ -17,14 +17,62 @@ typedef qint32 int32_t;
 // When changing, be carefull to not prevent the vectorizing
 // https://gcc.gnu.org/projects/tree-ssa/vectorization.html
 
+
+static inline bool useAlignedAlloc() {
+    // Will return true on MSVC 32 bit builds and falls for
+    // Linux 32 and 64 bit builds
+    return (sizeof(long double) == 8 && sizeof(CSAMPLE*) <= 8 &&
+            sizeof(CSAMPLE*) == sizeof(size_t));
+}
+
 // static
 CSAMPLE* SampleUtil::alloc(unsigned int size) {
```

Comment:
Yes good idea. We should switch to a Sample buffer class. With this, we can issue a bunch of problems. We can use optimized Sample functions that need not check if the start and end of the buffer is aligned. We can also fix the 160000 byte overallocation this helps to keep the buffers in the CPU cache. 
But the changes will effect a lot of files, so I like to do it in a separate PR after this one is merged.